src/engine/Match_search_mode.ml

```diff
@@ -617,6 +619,14 @@ let rec filter_ranges (env : env) (xs : (RM.t * MV.bindings list) list)
               *)
              | Some capture_bindings -> Some (r, capture_bindings @ new_bindings)
              )
+         | R.CondAnalysis (mvar, CondEntropyV2) -> begin
+              match !hook_pro_entropy_analysis with
+              | None -> error
```

Comment:
This *should* raise - if OSS is used on a rule that requires Pro analyzer, that rule should fail loudly, not silently.

I am not sure what error to use here - looking for similar behavior now.

Reply:
I think this is what I want? https://github.com/returntocorp/semgrep/blob/f46e887d6571b5cf2f899ea4a8d12bea6532d854/src/core/Rule.ml#L622C5-L622C18